### PR TITLE
Use a non-interactive progress writer in non-TTY environments

### PR DESCRIPTION
**Pull request by phamann**
Closes: #175 

### TL;DR
The current default progress writer uses interactive spinners on an event loop; while this style of output is useful in user terminal environments to give constant feedback of progression, it is not ideal in non-TTY environments, such as in continuous integration builds. 

### What?
This PR solves the CI issue by renaming the existing `text.QuietProgress` to `text.InteractiveProgress`, as it's not really "quiet", and then creates a new implementation of `text.QuietProgress` which only writes the step functions and not the intermediary writes to stdout. Lastly, we have a new constructor `text.NewProgress()` which is used throughout the program as a helper to abstract the logic which determines which progress should be used.

### CODE REVIEW

**Comment by phamann** [comment]
I've now fixed the failing tests 😄

**Review by Integralist — APPROVED** [review]
LGTM.
